### PR TITLE
Add mutation from `proc { }` to `lambda { }`

### PR DESCRIPTION
**Pull request by backus**

### CODE REVIEW

**Comment by backus** [comment]
@mbj this is ready for review

**Comment by backus** [comment]
@mbj this is ready for review

File: lib/mutant/ast/meta/send.rb [hunk 1]
```diff
@@ -21,6 +21,17 @@ class Send
 
         public :arguments
 
+        # Test if node is defining a proc
+        #
+        # @return [Boolean]
+        def proc?
+          selector.equal?(:proc) ||
```
Comment (mbj) [inline lib/mutant/ast/meta/send.rb]: Should this branch assert the receiver is implicit self?

Comment (backus) [inline lib/mutant/ast/meta/send.rb]: > Should this branch assert the receiver is implicit self?

Addressed

**Comment by backus** [comment]
This is rebased now

File: lib/mutant/ast/meta/send.rb [hunk 2]
```diff
@@ -21,6 +21,17 @@ class Send
 
         public :arguments
 
+        # Test if node is defining a proc
+        #
+        # @return [Boolean]
+        def proc?
+          !receiver && selector.equal?(:proc) ||
```
Comment (mbj) [inline lib/mutant/ast/meta/send.rb]: I think you should split this up into 2 private methods: `naked_proc?` and `proc_new?` or something.

File: lib/mutant/ast/meta/send.rb [hunk 3]
```diff
@@ -59,6 +66,25 @@ def receiver_possible_top_level_const?
...
+        end
+
+        # Test if node is `Proc.new { ... }`
+        #
+        # @return [Boolean]
+        def proc_new?
+          receiver                &&
+            selector.equal?(:new) &&
+            n_const?(receiver)    &&
+            Const.new(receiver).name.equal?(:Proc)
```
Comment (mbj) [inline lib/mutant/ast/meta/send.rb]: I'd love if we could at some point simply inject a value probe to evaluate the constant in the environment to simply do `#equal?(::Proc)`.

**Pull request merged**